src/particles/ImpactXParticleContainer.cpp

```diff
@@ -152,30 +152,36 @@ namespace impactx
         // When running without space charge and OpenMP parallelization,
         // we need to make sure we have enough tiles on level 0, grid 0
         // to thread over the available tiles. We try to set the tile_size
-        // appropriately here. The tiles start as (very large number, 8, 8)
-        // in (x, y, z). So we try to reduce the tile size in the y and z
-        // directions alternately until there are enough tiles for the number of threads.
+        // appropriately here. We start by setting the tile_size to the size of box 0.
+        // Then we try to reduce the tile size in the (z, y, x) directions alternately
+        // until there are enough tiles for the number of threads.
 
         const auto& ba = ParticleBoxArray(lid);
+        tile_size = ba[gid].size();
         auto n_logical = numTilesInBox(ba[gid], true, tile_size);
 
         int ntry = 0;
-        constexpr int max_tries = 6;
+        constexpr int max_tries = 10;
         while ((n_logical < nthreads) && (ntry++ < max_tries)) {
-            int idim = (ntry % 2) + 1;  // alternate between 1 and 2
+            int idim = 2 - (ntry % 3);  // alternate between (2, 1, 0)
+            if (tile_size[idim] == 1) { continue; }
             tile_size[idim] /= 2;
-            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(tile_size[idim] > 0,
-                                             "Failed to set proper tile size for the number of OpenMP threads. "
-                                             "Consider lowering the number of OpenMP threads via the environment variable OMP_NUM_THREADS."
-                                             );
             n_logical = numTilesInBox(ba[gid], true, tile_size);
         }
 
         if (n_logical < nthreads) {
-            amrex::Abort("ImpactParticleContainer::prepare() "
-                "could not find good tile size for the number of OpenMP threads. "
-                "Consider lowering the number of OpenMP threads via the environment variable OMP_NUM_THREADS."
-            );
+        ablastr::warn_manager::WMRecordWarning(
+        "ImpactxParticleContainer::prepare",
+        "Could not find good tile size for the number of OpenMP threads. "
+        "Lowering the number of OpenMP threads used compared to the environment variable OMP_NUM_THREADS. "
+        "This may result in poorer than expected performance. "
+        "You may want to try increasing the blocking factor and max grid size.",
+        ablastr::warn_manager::WarnPriority::medium
+        );
+
+#if defined(AMREX_USE_OMP)
+        omp_set_num_threads(n_logical);
```

Comment:
Two small changes we discussed:
- please include the `nthreads` number before and after we lower it (`n_logical`) in the warning message to the user
- please fix the alignment

Wording could be something like
```suggestion
        ablastr::warn_manager::WMRecordWarning(
        "ImpactxParticleContainer::prepare",
        "Could not find good tile size for the number of OpenMP threads (add: nthreads). "
        "Consider lowering the number of OpenMP threads via the environment variable OMP_NUM_THREADS. "
        "Will now lower it to the number of tiles, n_logical."
        "This may result in poorer than expected performance. "
        "You may want to try increasing the blocking factor and max grid size to create more tiles for OpenMP.",
        ablastr::warn_manager::WarnPriority::medium
        );

#if defined(AMREX_USE_OMP)
        omp_set_num_threads(n_logical);
```
In the wording, note that OMP_NUM_THREADS is not the only way we set the number of OMP threads. We pick a default by inspecting the system and also have an API to set it. `OMP_NUM_THREADS` is common and honored, unless the API is used to set num threads.